app/controllers/application_controller.rb

```diff
@@ -206,42 +206,29 @@ def needs_redirect?(host, domain)
   # 3. Otherwise nil
   #
   def self.default_community_fetch_strategy(domain)
-    # Find by domain
-    by_domain = Community.find_by_domain(domain)
+    raise ArgumentError("Domain cannot be nil.") if domain.nil?
 
-    if by_domain.present?
-      return by_domain
-    end
-
-    # Find by username
     app_domain = URLUtils.strip_port_from_host(APP_CONFIG.domain)
     ident = domain.chomp(".#{app_domain}")
-    by_ident = Community.where(ident: ident).first
 
-    if by_ident.present?
-      return by_ident
-    end
+    community = Community.where("domain = ? OR ident = ?", domain, ident).first
 
-    # If only one, use it
-    count = Community.count
+    return community if community.present?
 
-    if count == 1
-      return Community.first
-    end
+    # If only one, use it
+    return Community.first if Community.count == 1
 
     # Not found, return nil
     nil
   end
 
-  # Before filter to check if current user is the member of this community
-  # and if so, find the membership
   def fetch_community_membership
     if @current_user
-      if @current_user.communities.include?(@current_community)
-        @current_community_membership = CommunityMembership.find_by_person_id_and_community_id_and_status(@current_user.id, @current_community.id, "accepted")
-        unless @current_community_membership.last_page_load_date && @current_community_membership.last_page_load_date.to_date.eql?(Date.today)
-          Delayed::Job.enqueue(PageLoadedJob.new(@current_community_membership.id, request.host))
-        end
+      @current_community_membership = CommunityMembership.where(person_id: @current_user.id, community_id: @current_community.id, status: "accepted").first
+
+      last_page_load_date = Maybe(@current_community_membership).last_page_load_date.or_else(nil)
```

Comment:
Maybe is unneccessary here, as the next line guards against current_community_membership being nil.